test/unit/native_deps/cc_bin_uses_rust_library.cc

```diff
@@ -0,0 +1,7 @@
+extern "C" {
```

Comment:
Let's move this test from //test/unit/native_deps to //test/native_deps, I'd say it's not a unit test, it's assertion is that it can link.

Reply:
I worked with @hlopko over chat on this. There doesn't appear to be a way to write this test case so it disappears if clang isn't in use, so we're punting for now and removing the test. :/